### PR TITLE
Theming fixes

### PR DESCRIPTION
**Pull request by nickvergessen**
Fixes https://help.nextcloud.com/t/absender-von-freigabelinks/1620

Should backport to 9.0: #406 

@LukasReschke @MorrisJobke @schiessle

### CODE REVIEW

**Comment by MorrisJobke** [comment]
@nickvergessen Is this ready to review?

File: apps/dav/lib/Connector/PublicAuth.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ public function __construct(IRequest $request,
 		$this->session = $session;
 
 		// setup realm
-		$defaults = new \OC_Defaults();
+		$defaults = new \OCP\Defaults();
```
Comment (schiessle) [inline apps/dav/lib/Connector/PublicAuth.php]: any reasons why we don't use `\OC::$server->getThemingDefaults()` here and below?

Comment (nickvergessen) [inline apps/dav/lib/Connector/PublicAuth.php]: getThemingDefaults() is marked as internal, and the constructor of OCP\Default uses this internal method after this PR.

**Comment by nickvergessen** [comment]
Yes, ready for review

File: apps/theming/lib/template.php [hunk 2]
```diff
@@ -93,6 +97,15 @@ public function getSlogan() {
 		return $this->config->getAppValue('theming', 'slogan', $this->slogan);
 	}
 
+	public function getShortFooter() {
+		$slogan = $this->getSlogan();
+		$footer = '<a href="'. $this->getBaseUrl() . '" target="_blank"' .
```
Comment (icewind1991) [inline apps/theming/lib/template.php]: needs escaping?

Comment (nickvergessen) [inline apps/theming/lib/template.php]: No? Or atleast this is just copy paste of the parent class, so it would need fixing there as.

**Comment by MorrisJobke** [comment]
Tested and works 👍 Code looks good

**Comment by schiessle** [comment]
looks good :+1:

**Pull request merged**